module4/ex24_uniquish/ex24_uniquish.py

```diff
@@ -2,16 +2,35 @@ class Foo:
     def __init__(self, x):
         self.x = x
 
+    def __eq__(self, other):
+        if type(other) is type(self):
+            return self.x == other.x
+        return False
+
+    def __hash__(self):
+        return hash(self.x)
+
 
 class Uniquish:
-    pass
+    def __eq__(self, other):
+        if type(self) == type(other):
+            return self.__dict__ == other.__dict__
+
+    def __hash__(self):
+         return hash(tuple(sorted(self.__dict__.items())))
 
 
 class Bar(Uniquish):
     def __init__(self, x, y):
         self.x = x
         self.y = y
 
+    def __eq__(self, other):
```

Comment:
You don't have to redefine `__eq__` and `__hash__` methods, that's what `Uniquish` class inheritance is for.